internal/pkg/levee/testdata/src/example.com/tests/typeassert/tests.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeassert
+
+import (
+	"example.com/core"
+)
+
+func TestSourceAssertedFromTaintedEface(s core.Source) {
+	var e interface{} = s
+	core.Sink(e.(core.Source)) // want "a source has reached a sink"
+}
+
+func TestSourcePointerAssertedFromTaintedEface(s *core.Source) {
+	var e interface{} = s
+	core.Sink(e.(*core.Source)) // want "a source has reached a sink"
+}
```

Comment:
To distinguish between source detection and propagation detection here, consider adding

```golang
func Test(iface interface{}) {
  s := iface.(*core.Source)
  core.Sink(s) // want ...
}
```

Reply:
Very good idea.

This test will actually fail in the `interp` PR, because right it doesn't create new `Source`s from `TypeAssert`s, it only propagates taint through `typeassert`s. I'll go and fix that.

I've added an extra test that uses the very common `, ok` form.

(Interestingly, the `, ok` test passes, but the one you suggested doesn't. This is because in the `, ok` case we create a Source from the `Extract`, which we handle, while in the case you suggested we can only detect the source from the `TypeAssert` instruction, which we don't handle.